command/server/server.go

```diff
@@ -149,6 +155,42 @@ func (s *Server) setupMetrics(config *TelemetryConfig) error {
 	// Start system runtime metrics collection
 	go metrics.CollectProcessMetrics(3 * time.Second)
 
+	// Hook go-metrics into expvar on any /debug/metrics request, load all vars
+	// from the registry into expvar, and execute regular expvar handler.
+
+	if len(config.PrometheusAddr) != 0 {
+
+		prometheusMux := http.NewServeMux()
+
+		// this would cause a panic:
+		// panic: http: multiple registrations for /debug/vars
+		// http.HandleFunc("/debug/vars", e.expHandler)
+		// haven't found an elegant way, so just use a different endpoint
+
+		prometheusMux.HandleFunc("/debug/metrics", func(w http.ResponseWriter, r *http.Request) {
+			exp.ExpHandler(metrics.DefaultRegistry)
+		})
+
+		prometheusMux.HandleFunc("/debug/metrics/prometheus", func(w http.ResponseWriter, r *http.Request) {
+			prometheus.Handler(metrics.DefaultRegistry)
+		})
+
+		promServer := &http.Server{
+			Addr:           config.PrometheusAddr,
+			Handler:        prometheusMux,
+			ReadTimeout:    10 * time.Second,
```

Comment:
You can remove ReadTimeout, WriteTimeout and MaxHeaderBytes since I think they are populated by default by Go (please double check).